### PR TITLE
DOC-765: Added release notes for services updates

### PR DESCRIPTION
**Pull request by lnewson**
Related Ticket: DOC-765

Description of Changes:
* Added release notes for the service changes included in the 5.7 release.

Pre-checks:
- [x] Branch prefixed with `feature/` or `hotfix/`
- [x] `_data/nav.yml` has been updated (if applicable)
- [x] Files has been included where required (if applicable)
- [x] Files removed have been deleted, not just excluded from the build (if applicable)
- [x] (New product features only) Release Note added

Review:
- [x] Documentation Team Lead has reviewed
- [x] Product Manager has reviewed

### CODE REVIEW

**Comment by lnewson** [comment]
For reference, here's the difference in service WAR file size:

Before:
```
ephox-hyperlinking.war 53.1MB
ephox-image-proxy.war 59.9MB
ephox-spellchecking 55.3MB
```

After:
```
ephox-hyperlinking.war 35.8MB
ephox-image-proxy.war 30.7MB
ephox-spellchecking 39.7MB
```

**Review by EkimChau — APPROVED** [review]

File: release-notes/release-notes57.md [hunk 1]
```diff
@@ -187,6 +188,27 @@ The {{site.productname}} 5.7 release includes an accompanying release of the **P
...
+The {{site.productname}} 5.7 release includes accompanying changes affecting the {{site.productname}} **self-hosted** services for the following plugins:
+
+- The Enhanced Media Embed plugin (`mediaembed`)
+- The Image Tools plugin (`imagetools`)
+- The Link Checker plugin (`linkchecker`)
+- The Spell Checker Pro plugin (`tinymcespellchecker`)
+
+The Java server-side components (`ephox-spelling.war`, `ephox-hyperlinking.war`, and `ephox-image-proxy.war`) have been updated to **version 2.102**.
+
+This version upgraded and replaced internal dependencies to provide upstream bug fixes and smaller `.war` archives.
```
Comment (ephox-mogran) [inline release-notes/release-notes57.md:202]: smaller war archives wasn't really an aim, but the aims were complicated. Probably just "upgrading dependencies that were really out of date". I guess the war file size was sort of relevant in the moving to cats from scalaz, but the move wasn't really prompted by code size.

However, it's good enough. for release notes. Essentially "upgraded dependencies" ... the main driver being to modernise the codebase.

**Review by ephox-mogran — APPROVED** [review]

File: release-notes/release-notes57.md (hunk 1 above)
Comment (lnewson) [reply release-notes/release-notes57.md:202]: Yeah, I figured it wasn't the primary aim, however it was a side-effect that I thought was worth calling out :shrug:

**Review by ashfordneil — APPROVED** [review]

**Review by tylerkelly13 — CHANGES_REQUESTED** [review]
you forgot to update the `nav.yml` ;) otherwise, LGTM

**Review by tylerkelly13 — APPROVED** [review]

**Pull request merged**